salt/client/ssh/__init__.py

```diff
@@ -1085,6 +1099,13 @@ def run_wfunc(self):
                     )
             pillar_data = pillar.compile_pillar()
 
+            # Once pillar has been compiled, restore priority of minion opts
+            if self.fun not in master_centric_funcs:
+                log.debug('%s is a minion function', self.fun)
+                popts.update(opts_pkg)
```

Comment:
@cachedout I am still wondering if we need that `popts.update(opts_pkg)` first time. This one is a second.

Reply:
Will it do any harm though?

Reply:
I think this is fine and it makes sense.